src/request.rs

```diff
@@ -133,582 +125,415 @@ impl<'a> Request<'a> {
-                    arg.rdev,
-                    self.reply(),
-                );
-                #[cfg(feature = "abi-7-12")]
-                se.filesystem.mknod(
-                    self,
-                    self.request.nodeid(),
-                    &name,
-                    arg.mode,
-                    arg.umask,
-                    arg.rdev,
+                    self.request.nodeid().into(),
+                    &x.name(),
+                    x.mode(),
+                    x.umask(),
+                    x.rdev(),
                     self.reply(),
                 );
             }
-            ll::Operation::MkDir { arg, name } => {
-                #[cfg(not(feature = "abi-7-12"))]
-                se.filesystem.mkdir(
-                    self,
-                    self.request.nodeid(),
-                    &name,
-                    arg.mode,
-                    0,
-                    self.reply(),
-                );
-                #[cfg(feature = "abi-7-12")]
+            ll::Operation::MkDir(x) => {
                 se.filesystem.mkdir(
                     self,
-                    self.request.nodeid(),
-                    &name,
-                    arg.mode,
-                    arg.umask,
+                    self.request.nodeid().into(),
+                    x.name(),
+                    x.mode(),
+                    x.umask(),
                     self.reply(),
                 );
             }
-            ll::Operation::Unlink { name } => {
+            ll::Operation::Unlink(x) => {
                 se.filesystem
-                    .unlink(self, self.request.nodeid(), &name, self.reply());
+                    .unlink(self, self.request.nodeid().into(), x.name(), self.reply());
             }
-            ll::Operation::RmDir { name } => {
+            ll::Operation::RmDir(x) => {
                 se.filesystem
-                    .rmdir(self, self.request.nodeid(), &name, self.reply());
+                    .rmdir(self, self.request.nodeid().into(), x.name(), self.reply());
             }
-            ll::Operation::SymLink { name, link } => {
+            ll::Operation::SymLink(x) => {
                 se.filesystem.symlink(
                     self,
-                    self.request.nodeid(),
-                    &name,
-                    &Path::new(link),
+                    self.request.nodeid().into(),
+                    x.target(),
+                    &Path::new(x.link()),
                     self.reply(),
                 );
             }
-            ll::Operation::Rename { arg, name, newname } => {
+            ll::Operation::Rename(x) => {
                 se.filesystem.rename(
                     self,
-                    self.request.nodeid(),
-                    &name,
-                    arg.newdir,
-                    &newname,
+                    self.request.nodeid().into(),
+                    x.from().1,
+                    x.to().0.into(),
```

Comment:
I think all these tuples of (inode, name) will be easier to read if you introduce a struct with named fields, like you did for `CopyFileRangeFile`

Reply:
Done.  There may be scope to apply `struct FilenameInDir` more widely in the future to operations like `Lookup`, `Unlink`, etc.  I'll have a think about that.